### PR TITLE
Add filters and sort to mvlempyr

### PR DESCRIPTION
**Pull request by Soopyboo32**
Will be slower on first load but not much i can do about that :cry:

### CODE REVIEW

**Comment by Soopyboo32** [comment]
wait wtf why are the commits empty

**Comment by Soopyboo32** [comment]
:waytoodank:

**Comment by Palloxin** [comment]
> wait wtf why are the commits empty

itisinsideyou

File: proxy_server.js [hunk 1]
```diff
@@ -30,7 +31,36 @@ headers:`,
...
+    if (temp_cookies) curl += ` -H 'Cookie: ${temp_cookies}'`;
+    if (req.headers.origin2) curl += ` -H 'Origin: ${req.headers.origin2}'`;
+
+    console.log('Running curl command:', curl);
+
+    exec(
+      curl,
+      {
+        shell: 'C:\\Program Files\\git\\usr\\bin\\bash.exe',
+      },
```
Comment (nyagami) [inline proxy_server.js:46]: There are linux/macos devs. Also I dont install git in default location.

File: proxy_server.js [hunk 2]
```diff
@@ -30,7 +31,36 @@ headers:`,
...
+      (error, stdout, stderr) => {
+        if (error) {
+          console.error(`exec error: ${error}`);
+          return;
+        }
+        if (stderr) {
+          console.error(`stderr: ${stderr}`);
+        }
```
Comment (nyagami) [inline proxy_server.js:54]: Can we send the error back to client?

File: src/components/accordions/Headers/index.tsx [hunk 3]
```diff
@@ -62,6 +82,27 @@ export default function HeadersSection() {
...
+      >
+        Fetch Mode:
+        <Select
+          variant="outlined"
+          label="Fetch Mode"
+          value={fetchMode}
+          onChange={e => setFetchMode(e.target.value)}
+        >
+          <MenuItem value="proxy">Proxy</MenuItem>
+          <MenuItem value="cookie">Cookie</MenuItem>
```
Comment (nyagami) [inline src/components/accordions/Headers/index.tsx:102]: I think options should be types of fetcher: proxy, node fetch and curl

File: src/plugins/english/mvlempyr.ts [hunk 4]
```diff
@@ -1,22 +1,22 @@
...
 
   headers = {
     'origin': 'https://www.mvlempyr.com/',
   };
 
-  _pageCache = new Map<string, string>();
-  _chapSite = 'https://chp.mvlempyr.com/';
-  _assetsSite = 'https://assets.mvlempyr.com/';
-  _allNovels: WPData[] | undefined;
+  _chapSite = 'https://chp.mvlempyr.net/';
```
Comment (Soopyboo32) [inline src/plugins/english/mvlempyr.ts:17]: not fully sure what the difference between the .net and .com site are, but some novels dont work on .com :shrug:

File: src/components/accordions/Headers/index.tsx [hunk 5]
```diff
@@ -1,12 +1,21 @@
...
+  TextField,
+} from '@mui/material';
 import useDebounce from '@hooks/useDebounce';
 
 export default function HeadersSection() {
   const [loading, setLoading] = useState(false);
   const [cookies, setCookies] = useState('');
   const debounceCookies = useDebounce(cookies, 500);
+  const [fetchMode, setFetchMode] = useState('proxy');
+  const debounceFetchMode = useDebounce(fetchMode, 500);
```
Comment (nyagami) [inline src/components/accordions/Headers/index.tsx:18]: no need debounce

File: src/components/accordions/Headers/index.tsx [hunk 6]
```diff
@@ -62,6 +82,27 @@ export default function HeadersSection() {
...
+      >
+        Fetch Mode:
+        <Select
+          variant="outlined"
+          label="Fetcher"
+          value={fetchMode}
+          onChange={e => setFetchMode(e.target.value)}
+        >
+          <MenuItem value="proxy">Proxy</MenuItem>
+          <MenuItem value="cookie">Node fetch</MenuItem>
```
Comment (nyagami) [inline src/components/accordions/Headers/index.tsx:102]: should change value to not confusing people

File: src/components/accordions/Headers/index.tsx [hunk 7]
```diff
@@ -23,6 +32,17 @@ export default function HeadersSection() {
...
+      method: 'POST',
+      body: debounceFetchMode,
+    })
+      .then(() => setAlertVisble(true))
+      .finally(() => setLoading(false));
+  }, [debounceFetchMode]);
+
   return (
     <AccordionContainer title="Headers" loading={loading}>
       {alertVisible ? (
```
Comment (nyagami) [inline src/components/accordions/Headers/index.tsx:48]: Message is not correct when changing fetch mode

**Comment by nyagami** [comment]
Im not able to run curl fetch
image

File: src/plugins/english/mvlempyr.ts [hunk 8]
```diff
@@ -1,22 +1,22 @@
...
 class MVLEMPYRPlugin implements Plugin.PluginBase {
   id = 'mvlempyr.com';
   name = 'MVLEMPYR';
   icon = 'src/en/mvlempyr/icon.png';
   site = 'https://www.mvlempyr.com/';
-  version = '1.0.0';
+  version = '1.0.1';
 
   headers = {
     'origin': 'https://www.mvlempyr.com/',
```
Comment (nyagami) [inline src/plugins/english/mvlempyr.ts:14]: no need origin header, 
my working headers for fetching tags in parseNovel 
```js
 headers: {
          'accept': '*/*',
          'Referer': 'https://www.mvlempyr.com/',
        },
```

**Comment by Soopyboo32** [comment]
> Im not able to run curl fetch image

If u have git bash installed at a non-default location u gotta set then env var `BASH_LOCATION` to use curl

**Review by nyagami — APPROVED** [review]

**Pull request merged**